collector/cpu_linux.go

```diff
@@ -276,6 +299,39 @@ func (c *cpuCollector) updateThermalThrottle(ch chan<- prometheus.Metric) error
 	return nil
 }
 
+func contains(s []uint16, e uint16) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+// updateStat reads /proc/stat through procfs and exports CPU-related metrics.
+func (c *cpuCollector) updateIsolated(ch chan<- prometheus.Metric) error {
+	stats, err := c.fs.Stat()
+	if err != nil {
+		return err
+	}
+
+	c.updateCPUStats(stats.CPU)
```

Comment:
I don't think this nor locking the mutex is necessary.
There is also no need to use contains(), I'd just do:
```
for _, cpu := range c.isolatedCpus {
  ch <- prometheus.MustNewConstMetric(c.cpuIsolated, prometheus.GaugeValue, 1.0, cpu)
```

IMO we don't need a timeseries with 0.0 for the cpus that are not isolated.